### PR TITLE
implementation ChromBackend

### PR DESCRIPTION
**Pull request by philouail**
So there is still a lot lot to do but as discussed I am trying to reduce the size of the PR. I hope that does not make it too confusing.

The big changes are the clear separation between what is a peaksVariables and what is a chromvariables. 
See `ChromBackend-functions.R`

Also implementation of respective function for each such as `validPeaksData()` and `validChromData()`

Quick question:
- I added a replacement method for chromIndex, is that okay or in general we don't want people to change that ?
- is it a problem that `chromData()` returns a `data.frame` and not `DataFrame` ? I don't really see the issue for now..

Next I will add the `ChromBackendMemory()` implementation, i have the code, just need to clean it.
Also will look into some maintainer stuff, if you have advices don't hesitate.

### CODE REVIEW

**Comment by philouail** [comment]
I changed the .yml file, the checks should not fail on warnings anymore :)

File: R/ChromBackend.R
Comment (jorainer) [inline R/ChromBackend.R:137]: I would suggest you also describe the `drop` parameter here to clearly define what `drop = TRUE` and `drop = FALSE` is expected to do. Same also for the `peaksData()` documentation below. And you also need to have a `@param drop` documentation entry (maybe you have it already but I did not see it here). We need to be very clear with this parameter, especially because we don't have that in *Spectra*.

**Review by jorainer — CHANGES_REQUESTED** [review]
Looks good Phili! Thanks! Just some minor change requests.

Regarding the question if `chromData()` should return a `DataFrame` or `data.frame`, I don't see a problem for `data.frame` at the moment - so, given that it is also slightly more efficient, I would maybe change to `data.frame` instead of `DataFrame`.

File: R/ChromBackend-functions.R
Comment (philouail) [reply R/ChromBackend-functions.R:20]: This is used later on for the filter function that you implemented. I kept it on another file locally, but I only wanted things related to what is implemented now to be in this PR. So it will probably come back when I start implementing filter functions :)

File: R/AllGenerics.R
Comment (philouail) [reply R/AllGenerics.R:55]: mmm not that I can see ! It's also in the `AllGenerics.R` in `Spectra` :)

File: R/ChromBackend-functions.R
Comment (philouail) [reply R/ChromBackend-functions.R:142]: mmm follow up question then, should the setValidity() be fully removed for `ChromBackend` ? And we make sure we properly use  validChromData() and validPeaksData() when necessary ?

**Comment by philouail** [comment]
Hi so I answered / addressed the comments, thanks for the advices ! I went through the documentation a bit more realised I missed removing some stuff also.
 tell me if there is something else :) and for moving the `reset()` to `ProtGenerics` also

File: R/AllGenerics.R
Comment (jorainer) [reply R/AllGenerics.R:55]: OK - we might then need (later at some point) add that to *ProtGenerics*.

File: R/ChromBackend-functions.R
Comment (jorainer) [reply R/ChromBackend-functions.R:20]: Thanks for the clarification!

File: R/ChromBackend-functions.R
Comment (jorainer) [reply R/ChromBackend-functions.R:147]: Good question. I would be OK with removing it for now and then, if we see we need it, add it back. You, as package maintainer, can decide if you want to remove or not ;)

File: R/ChromBackend.R
Comment (jorainer) [inline R/ChromBackend.R:139]: Please reformulate here *... as default. this mean...* - maybe with * ... as default. With `drop = FALSE` the method should return ...*.

File: R/ChromBackend.R
Comment (jorainer) [inline R/ChromBackend.R:152]: See previous comment.

**Review by jorainer — APPROVED** [review]
Looks good now, thanks Phili!

Before merging make sure you fix the two comments/issues in the documentation.

Regarding `reset()` - yes, we might want to add that to *ProtGenerics*, but for now define it also here. We can change later.

**Pull request merged**
